examples/src/main/java/org/apache/spark/examples/streaming/JavaDirectKerberizedKafkaWordCount.java

```diff
@@ -0,0 +1,118 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.spark.examples.streaming;
+
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.Arrays;
+import java.util.Map;
+import java.util.Set;
+import java.util.regex.Pattern;
+
+import scala.Tuple2;
+
+import org.apache.kafka.clients.CommonClientConfigs;
+import org.apache.kafka.common.security.auth.SecurityProtocol;
+import org.apache.kafka.clients.consumer.ConsumerConfig;
+import org.apache.kafka.clients.consumer.ConsumerRecord;
+import org.apache.kafka.common.serialization.StringDeserializer;
+
+import org.apache.spark.SparkConf;
+import org.apache.spark.streaming.api.java.*;
+import org.apache.spark.streaming.kafka010.ConsumerStrategies;
+import org.apache.spark.streaming.kafka010.KafkaUtils;
+import org.apache.spark.streaming.kafka010.LocationStrategies;
+import org.apache.spark.streaming.Durations;
+
+/**
+ * Consumes messages from one or more topics in Kafka and does wordcount.
+ * Usage: JavaDirectKerberizedKafkaWordCount <brokers> <groupId> <topics>
+ *   <brokers> is a list of one or more Kafka brokers
+ *   <groupId> is a consumer group name to consume from topics
+ *   <topics> is a list of one or more kafka topics to consume from
+ *
+ * Example:
+ *    $ bin/run-example --files ${path}/kafka_jaas.conf \
+ *      --driver-java-options "-Djava.security.auth.login.config=${path}/kafka_jaas.conf" \
+ *      --conf \
+ *      "spark.executor.extraJavaOptions=-Djava.security.auth.login.config=./kafka_jaas.conf" \
+ *      streaming.JavaDirectKerberizedKafkaWordCount broker1-host:port,broker2-host:port \
+ *      consumer-group topic1,topic2
+ *
+ * kafka_jaas.conf can manually create, template as:
+ *   KafkaClient {
+ *     com.sun.security.auth.module.Krb5LoginModule required
+ *     keyTab="${path_of_keytab}/kafka.service.keytab"
```

Comment:
Since the keytab file is not added to the `--files` section it must exist on every server. Worth to mention as prerequisite since it's not obvious for everybody.

Reply:
> Since the keytab file is not added to the `--files` section it must exist on every server. Worth to mention as prerequisite since it's not obvious for everybody.

As cmd example show, `--file` should used.

Reply:
I'm fine to add keytab file to the `--files` section but then the keytab path in the jaas file has to be modified to `./kafka.service.keytab` since `--files` doesn't preserve the path.

Reply:
@gaborgsomogyi you are right, in my environment, keytab in same path of each node. I think we can two jaas file:
for driver `keyTab="${path_of_keytab}/kafka.service.keytab"`
for executor `keyTab="./kafka.service.keytab"`
and run-example as 
`bin/run-example --files ${path}/kafka_executor_jaas.conf,${kyetab_path}/kafka.service.keytab \
  --driver-java-options "-Djava.security.auth.login.config=${path}/kafka_driver_jaas.conf" \
  --conf \
  "spark.executor.extraJavaOptions=-Djava.security.auth.login.config=./kafka_executor_jaas.conf" \
  streaming.JavaDirectKerberizedKafkaWordCount broker1-host:port,broker2-host:port \
  consumer-group topic1,topic2`
Do you have any succinct advice?

Reply:
Your last example is almost good but it will work only in client mode. In cluster mode the driver runs on a random machine where maybe the keytab not exists. I would suggest this:
```
bin/run-example --files ${jaas_path}/jaas.conf,${kyetab_path}/kafka.service.keytab \
--driver-java-options "-Djava.security.auth.login.config=./jaas.conf" \
--conf "spark.executor.extraJavaOptions=-Djava.security.auth.login.config=./jaas.conf" \
streaming.JavaDirectKerberizedKafkaWordCount broker1-host:port,broker2-host:port \
consumer-group topic1,topic2
```
and in the jaas file:
```
...
  keyTab="./kafka.service.keytab"
...
```

Reply:
> Your last example is almost good but it will work only in client mode. In cluster mode the driver runs on a random machine where maybe the keytab not exists. I would suggest this:
> 
> ```
> bin/run-example --files ${jaas_path}/jaas.conf,${keytab_path}/kafka.service.keytab \
> --driver-java-options "-Djava.security.auth.login.config=./jaas.conf" \
> --conf "spark.executor.extraJavaOptions=-Djava.security.auth.login.config=./jaas.conf" \
> streaming.JavaDirectKerberizedKafkaWordCount broker1-host:port,broker2-host:port \
> consumer-group topic1,topic2
> ```
> 
> and in the jaas file:
> 
> ```
> ...
>   keyTab="./kafka.service.keytab"
> ...
> ```

Yes, cluster is diff form client. It need a different cmd.